### PR TITLE
Update for TS 5.6

### PR DESCRIPTION
**Pull request by jakebailey**
I had most of this change locally already while testing a 5.7 PR.

### CODE REVIEW

File: packages/tsconfig-reference/tsconfig.tsbuildinfo
Comment (jakebailey) [inline packages/tsconfig-reference/tsconfig.tsbuildinfo:1]: Ah, these files are now generated

File: packages/documentation/copy/en/project-config/Compiler Options.md [hunk 1]
```diff
@@ -54,75 +54,66 @@ tsc app.ts util.ts --target esnext --outfile index.js
...
   <td><p><code>boolean</code></p>
 </td>
 </tr>
-<tr class="option-description even"><td colspan="3">
+<tr class="option-description odd"><td colspan="3">
 <p>Initializes a TypeScript project and creates a tsconfig.json file.</p>
 </td></tr>
 
-<tr class='odd' name='listFilesOnly'>
+<tr class='even' name='listFilesOnly'>
```
Comment (DanielRosenwasser) [inline packages/documentation/copy/en/project-config/Compiler Options.md:75]: Not in this PR but one day https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child

**Review by DanielRosenwasser — APPROVED** [review]

File: packages/tsconfig-reference/scripts/schema/result/schema.json [hunk 2]
```diff
@@ -834,12 +840,13 @@
                       "esnext.decorators",
                       "esnext.object",
                       "esnext.regexp",
+                      "esnext.iterator",
                       "decorators",
                       "decorators.legacy"
                     ]
                   },
                   {
```
Comment (navya9singh) [inline packages/tsconfig-reference/scripts/schema/result/schema.json:848]: Just for my understanding, could you tell me what these pattern changes mean?

Comment (jakebailey) [reply packages/tsconfig-reference/scripts/schema/result/schema.json:848]: These options are case insensitive, so this is just the absolutely ridiculous regex to verify the union of a bunch of strings

Comment (jakebailey) [reply packages/tsconfig-reference/scripts/schema/result/schema.json:848]: In a better world, the regex syntax would accept syntax like `(?i)`, but JS doesn't have that syntax, it puts it _outside_ the regex expression and a load of consuming tools do not parse out strings like `"/blah/i"`

**Review by navya9singh — APPROVED** [review]

**Pull request merged**